app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailFragment.kt

```diff
@@ -0,0 +1,723 @@
+        fun addPhoto(){
+            addPhotoSearchForResult.launch("image/*")
+        }
+
+        fun openPhoto(index: Int) {
+            setFragmentResultListener("REQUEST_KEY"){
+                resultKey, bundle -> if(resultKey == "REQUEST_KEY"){
+                    val photoToDelete = bundle.getInt("DELETE_PHOTO_INDEX")
+                    viewModel.deletePhoto(photoToDelete)
+                }
+            }
+            navController.navigate(
+                EventDetailFragmentDirections
+                    .actionEventDetailFragmentToPhotoDetailFragment(
+                        viewModel.photos.value[index].toString(), index
+                    )
+            )
+        }
+
+        binding.apply {
+            collectLatestLifecycleFlow(viewModel.photos) { photoList ->
+                addPhotoLayout.isVisible = !(isAttendee && photoList.isEmpty())
+
+                val adapter = PhotoItemAdapter(
+                    photoList,
+                    onPhotoClick = { index -> openPhoto(index) },
+                    onAddPhotoClick = { addPhoto() },
+                    userIsAttendee = isAttendee
+                )
+                photosRecyclerView.adapter = adapter
+                photosRecyclerView.layoutManager = LinearLayoutManager(
+                    requireContext(), LinearLayoutManager.HORIZONTAL, false
+                )
+
+                addPhotoPlusSign.isVisible = photoList.isEmpty()
+                addPhotoTextView.isVisible = photoList.isEmpty()
+                photosTextView.isVisible = photoList.isNotEmpty()
+                photosRecyclerView.isVisible = photoList.isNotEmpty()
+
+            }
+            addPhotoTextView.setOnClickListener {
+                addPhoto()
+            }
+            addPhotoPlusSign.setOnClickListener {
+                addPhoto()
+            }
+        }
+    }
+
+    private fun setupStartTimeLayout(){
+        startTimeDateBinding.apply {
+
+            timeAndDateBeginningText.text = getString(R.string.from)
+
+            collectLatestLifecycleFlow(viewModel.selectedStartTime) { selectedStartTime ->
+                timeTextView.text = selectedStartTime.format(
+                    DateTimeFormatter.ofPattern("HH:mm")
+                )
+            }
+            collectLatestLifecycleFlow(viewModel.selectedStartDate) { selectedStartDate ->
+                dateTextView.text = selectedStartDate.format(
+                    DateTimeFormatter.ofPattern("MMM dd yyyy")
+                )
+            }
+
+            if (!isAttendee) {
+                collectLatestLifecycleFlow(viewModel.isInEditMode) { isEditing ->
+                    timeTextView.isEnabled = isEditing
+                    timeButton.isEnabled = isEditing
+                    timeButton.isVisible = isEditing
+                    dateTextView.isEnabled = isEditing
+                    dateButton.isEnabled = isEditing
+                    dateButton.isVisible = isEditing
+                }
```

Comment:
I'd personally prefer not structuring setting up your UI based on the sections of your layout, but rather based on the states you observe. Here, you'll add multiple observers to `isInEditMode`, since you call `collectLatestLifecycleFlow` at multiple places. One observer would be enough though. I don't really see a reason at all why you need to put setting up every section into its own function, I'd just have one `setupObservers()` or so in which you add all your flow collectors. That still makes it easy to find stuff.